src/deluge/gui/menu_item/runtime_feature/settings.h

```diff
@@ -23,13 +23,14 @@
 
 namespace deluge::gui::menu_item::runtime_feature {
 
-class Settings final : public Submenu<RuntimeFeatureSettingType::MaxElement> {
+/// Some runtime feature settings are squirrled away in submenus.
+constexpr size_t kNonTopLevelSettings = 3;
```

Comment:
Can this be automatically derived from the instantiating array instead?

Reply:
@bobtwinkles provided me with this code and I believe he hardcoded the number as well. I think it becomes complicated when we start having multiple sub menus.

The issue is the menu items are all enumerated together and when you have a sub menu you have to reduce the total number of items in the main menu.

Any suggestion on how to do this? As this number will need to be the sum of all sub menu items from all future sub menus.

Reply:
If/when we end up adding more this should be broken in to separate constants for each submenu that are summed together. The trouble with deriving it from the instantiating array is that the definition probably should live in the `.cpp` file which makes it annoying to access here - we could declare it in the header and pull the size in here instead, which is maybe better? Something like

```cpp
extern std::array<MenuItem*, 4> automationSubmenu;

constexpr size_t kNonToLevelSettings = (automationSubmenu.size() - 1);
```

(untested, may not even compile). I seem to remember Sean running in to trouble with using an externally defined array for the submenu though I'm not sure why.

Reply:
yea tried your code @bobtwinkles and it won't compile